### DISCUSSION TITLE
Unable to resolve imports(DSCStablecoin)

### DISCUSSION THREAD

**Discussion by naveenPraakash** [Cyfrin/foundry-full-course-cu#1061] [Q&A] [answered]
Hey everybody, I am getting the following error while trying to run forge build  in the DSCStableCoin Project, can someone please help me with this.

error :

 Unable to resolve imports:
      "../../script/DeployDSC.s.sol" in "/home/naveen/solidity-course/foundry-f23/stablecoin/test/DSCEngineTest.t.sol"
with remappings:
      @chainlink/contracts/=/home/naveen/solidity-course/foundry-f23/stablecoin/lib/chainlink-brownie-contracts/contracts/
      @openzeppelin/contracts/=/home/naveen/solidity-course/foundry-f23/stablecoin/lib/openzeppelin-contracts/contracts/
      chainlink-brownie-contracts/=/home/naveen/solidity-course/foundry-f23/stablecoin/lib/chainlink-brownie-contracts/
      ds-test/=/home/naveen/solidity-course/foundry-f23/stablecoin/lib/forge-std/lib/ds-test/src/
      forge-std/=/home/naveen/solidity-course/foundry-f23/stablecoin/lib/forge-std/src/
[⠑] Unable to resolve imports:/home/naveen/solidity-course/foundry-f23/stablecoin/lib/openzeppelin-contracts/
code snippet : ( Some parts have been commented to identify the specific error)
``` 
pragma solidity ^0.8.18;

import {Test} from "forge-std/Test.sol";
import {DeployDSC} from "../../script/DeployDSC.s.sol";

//import {DecentralizedStablecoin} from "../../src/DecentralizedStablecoin.sol";

// import {DSCEngine} from "../../src/DSCEngine.sol";
// import {HelperConfig} from "../../script/HelperConfig.s.sol";
// import {ERC20Mock} from "@openzeppelin/contracts/mocks/ERC20Mock.sol";

contract DSCEngineTest is Test {
    DeployDSC deployer;
    //DecentralizedStablecoin dsc;
    //DSCEngine dsce;
    //HelperConfig config;
    address ethUsdPriceFeed;
    address weth;

    address public USER = makeAddr("user");
    uint256 public constant AMOUNT_COLLATERAL = 10 ether;
    uint256 public constant STARTING_ERC20_BALANCE = 100 ether;

    function setUp() public {
        deployer = new DeployDSC();

        //(dsc, dsce, config) = deployer.run();
        // (ethUsdPriceFeed, , weth, , ) = config.activeNetworkConfig();

        // ERC20Mock(weth).mint(USER, STARTING_ERC20_BALANCE);
    }
}
```
foundry.toml

```
[profile.default]
src = "src"
out = "out"
libs = ["lib"]
remappings = [
    '@chainlink/contracts/=lib/chainlink-brownie-contracts/contracts/',
    '@openzeppelin/contracts=lib/openzeppelin-contracts/contracts',
]
```

**Comment by Niferu** [comment]
Hi @naveenPraakash 

Can you link your repo?

**Reply by naveenPraakash** [reply]
Hi @Neftyr thankyou for replying, the issue seems to have been resolved as all my tests are passing however I am still getting this "Unable to resolve imports" warning thing on compilation. Please find the link for the repo below for reference : https://github.com/naveenPraakash/decentralizedStablecoin-DSC-

**Reply by Niferu** [reply, marked as answer]
Hi @naveenPraakash 

In your `DSCEngineTest.t.sol` you are using below:

```solidity
import {DeployDSC} from "../../script/DeployDSC.s.sol";
import {DecentralizedStablecoin} from "../../src/DecentralizedStablecoin.sol";
import {DSCEngine} from "../../src/DSCEngine.sol";
import {HelperConfig} from "../../script/HelperConfig.s.sol";
```

But your folder structure is different and using `../` means getting back 1 folder up, so in my opinion you should be using below, as you have your test contract under test folder only

```solidity
import {DeployDSC} from "../script/DeployDSC.s.sol";
import {DecentralizedStablecoin} from "../src/DecentralizedStablecoin.sol";
import {DSCEngine} from "../src/DSCEngine.sol";
import {HelperConfig} from "../script/HelperConfig.s.sol";
```

**Reply by naveenPraakash** [reply]
Hi @Neftyr thankyou for your response , I understood , I messed up my folder structure a bit. Now it is'nt showing any warnings, but I am a bit confused, why did the code compile with the error in place( although there was a warning ), and how did all my test pass (the setup itself should have failed).

**Reply by Niferu** [reply]
Hi @naveenPraakash 

Yea indeed thats pretty bizzare why you did not get errors like imports unresolved, anyway good that everything is working now and glad I could help :)
